chia/consensus/block_header_validation.py

```diff
@@ -198,6 +198,8 @@ def validate_unfinished_header_block(
                         icc_iters_proof,
                         sub_slot.infused_challenge_chain.infused_challenge_chain_end_of_slot_vdf.output,
                     )
+                    if icc_iters_committed is None:
+                        return None, ValidationError(Err.INVALID_ICC_EOS_VDF)
```

Comment:
`icc_iters_committed` may not be `None` in the `replace()` call below

Reply:
this line is not covered by tests. It could probably have been an `assert`, but it seemed a bit more conservative to treat it as a consensus failure

Reply:
Yeah, I noticed you didn't pile it into the existing asserts and I appreciate the consideration of what you did add.  The structure just feels like we're missing something else to identify what case we're in etc.  But, I expect that even if I am correct that addressing that situation would be entirely separate from this PR.  Thanks.

Reply:
we have some non-trivial invariants of the optionals in our FulBlock